examples/gcoap/gcoap_cli.c

```diff
@@ -144,7 +145,8 @@ static void _resp_handler(unsigned req_state, coap_pkt_t* pdu,
             block.blknum++;
             coap_opt_add_block2_control(pdu, &block);
             int len = coap_opt_finish(pdu, COAP_OPT_FINISH_NONE);
-            gcoap_req_send((uint8_t *)pdu->hdr, len, remote, _resp_handler);
+            gcoap_req_send((uint8_t *)pdu->hdr, len, remote,
+                           _resp_handler, NULL);
```

Comment:
I would set the context for the new request with the context from the received response. I realize it's NULL at present, but this code should not make that assumption.

Reply:
of course, makes a lot of sense!